common/config/rush/common-versions.json

```diff
@@ -1,58 +0,0 @@
-/**
- * This configuration file specifies NPM dependency version selections that affect all projects
- * in a Rush repo.  For full documentation, please see https://rushjs.io
- */
-{
-  "$schema": "https://developer.microsoft.com/json-schemas/rush/v5/common-versions.schema.json",
-  /**
-   * A table that specifies a "preferred version" for a dependency package. The "preferred version"
-   * is typically used to hold an indirect dependency back to a specific version, however generally
-   * it can be any SemVer range specifier (e.g. "~1.2.3"), and it will narrow any (compatible)
-   * SemVer range specifier.  See the Rush documentation for details about this feature.
-   */
-  "preferredVersions": {
-    /**
-     * When someone asks for "^1.0.0" make sure they get "1.2.3" when working in this repo,
-     * instead of the latest version.
-     */
-    // "some-package": "1.2.3"
-
-    // Pinning a dependency here forces it to show up in the generated common/temp/package.json,
-    // which reduces the chance of duplicates being added to the dependency tree and can help fix
-    // issues with missing peer dependencies.
-    "@types/react": "16.8.11",
-    "@types/react-dom": "16.8.4",
-    "@types/webpack-env": "1.13.9",
-    "css-loader": "^0.28.7",
-    // enzyme@3.10 causes issues with asAsync test in utilities
-    "enzyme": "~3.9.0",
-    "jest": "24.1.0",
-    // React deps are pinned to prevent Invalid Hook Warnings with duplicate React deps.
-    // https://reactjs.org/warnings/invalid-hook-call-warning.html#duplicate-react
-    // NOTE: If these are changed, the React versions in most index.html files (and EmptyIndexHtml.mustache)
-    // should also change.
-    "react": "16.8.6",
-    "react-dom": "16.8.6",
```

Comment:
Also related to React Hooks volatility.. any versions of React hardcoded in various HTML files and other files in our repo need to match the primary dependency.

Reply:
I think yarn will handle this one as long as the truth is inside package.json (the range listed there seems okay) and the hoisting makes sure there's only one copy.

Reply:
What happens if we have another scenario where different packages need different versions of something?

Reply:
You would just specify that in package.json like normal? So, you're pointing out the need to have a validator & some sort of exception list for said validator. I'm hoping we will get to adding a level of validation that involves the `depcheck` module and some rules we invent for our repo:

https://www.npmjs.com/package/depcheck